### ISSUE TITLE
Read analysed data from jdx file that store Chemspectra data

### ISSUE THREAD

**Issue by baolanlequang** [NFDI4Chem/nmrium-react-wrapper#41]
@CS76 
## Current feature of NMRium wrapper
- Jcamp files (*.jdx files) are read default spectra data and meta data content.

## Suggestion or discussion
- Can we support that transfer or read custom fields on jcamp files like peaks, multiplicity, and integrals?

**Comment by baolanlequang** [comment]
In jcamp file which is generated by Chemspectra, there are fields in the file representing integration, multiplicity, and peaks.

#### 1. Integration
The data for integration is stored in the file after this line `##$OBSERVEDINTEGRALS= (X Y Z)`
The format of this data is:
```
(<lowest value of x>, <highest value of x>,  <reference area>, <absolute area>)
```
Example:
```
##$OBSERVEDINTEGRALS= (X Y Z)
(117.54850535600065, 167.7517328787416, 0.7688874662387057, 7458045956.1968775)
```

#### 2. Multiplicity
The data for multiplicity is stored in the file after this line `##$OBSERVEDMULTIPLETS=`
The format of this data is:
```
(<index of multiplicty>, <lowest value of x>, <highest value of x>, <center of multiplicty>,  <area>, <index of multiplicty for peaks reference>, <type of multiplicty>, <js of multiplicity>)
```
Example:
```
##$OBSERVEDMULTIPLETS=
(1, 48.49425617027523, 98.8959225932613, 71.51395088422314, 1.0, 1, m, A)
```

The peaks data of multiplicity is stored in the file after this line `##$OBSERVEDMULTIPLETSPEAKS=`
The format of this data is:
```
(<index of multiplicity>, <x value of peak>, <y value of peak>)
```
Example:
```
##$OBSERVEDMULTIPLETS=
##$OBSERVEDMULTIPLETSPEAKS=
(1, 88.42100629180075, 41460686.26228226)
(1, 83.37378114407994, 47730911.617038734)
(1, 61.231857165231986, 4610227.459403367)
(1, 53.02915893577986, 32141353.241591156)
```

#### 3. Peaks data
The data of peaks is stored in the file at sections `$$ === CHEMSPECTRA PEAK TABLE EDIT ===` and `$$ === CHEMSPECTRA PEAK TABLE AUTO ===` for used edited peaks and peaks was automatic generated by Chemspectra, respectively.

The data of peak is stored as JCAMP-DX standard.
```
Example:
```
$$ === CHEMSPECTRA PEAK TABLE AUTO ===
##TITLE=HMS-Q303_11
##JCAMP-DX=5.00
##DATA TYPE=NMRPEAKTABLE
##DATA CLASS=PEAKTABLE
##$CSCATEGORY=AUTO_PEAK
##$CSTHRESHOLD=0.005
##MAXX=235.4798379479265
##MAXY=297871004.65036833
##MINX=-15.479837947926484
##MINY=-9603659.060932525
##NPOINTS=100
##PEAKTABLE= (XY..XY)
76.71818379450579, 297871004.65036833
77.35387376000023, 287795952.81256557
```

An example of the generated file. Note: because of limitation of github supported file, I compressed it as zip, you can extract it.
[HMS-Q303_11.edit.zip](https://github.com/NFDI4Chem/nmrium-react-wrapper/files/9494956/HMS-Q303_11.edit.zip)

**Comment by CS76** [comment]
Update: JCAMP files generated from ChemSpectra are now handed over to the nmrium team. It needs to be tested once an update is released on nmrium.

**Comment by NRayya** [comment]
Related to https://github.com/cheminfo/nmrium/issues/2039

**Comment by baolanlequang** [comment]
@NRayya To solve our issue, I've implemented new functions to convert files from Chemspectra to `*.nmrium` format at https://github.com/ComPlat/chem-spectra-app/pull/55, but it needs to be tested.

**Comment by baolanlequang** [comment]
@NRayya This function will be released on ChemotionELN version 1.5

**Comment by NRayya** [comment]
Feature implemented and closing the issue.
